client/src/components/Login.js

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import axios from "axios";
+import { Button, Form, FormGroup, Label, Input, Spinner } from "reactstrap";
+
+class Login extends React.Component {
+  state = {
+    username: "",
+    password: "",
+  };
+
+  handleInputChange = event => {
+    const { name, value } = event.target;
```

Comment:
There are a few alternative ways you could use destructuring here to the same effect:
- taking target off of the event parameter (so as to slightly reduce repetition)
```javascript
handleInputChange = ({ target }) => {
  const { name, value } = target;
  this.setState({
    [name]: value
  })
};
```
- or, take off name and value since you don't need target (note that you _won't_ have access to the target object here, but only the innermost properties you destructure off, in this case name and value):
```javascript
handleInputChange = ({ target: { name, value } }) => {
  this.setState({
    [name]: value
  })
};
```
nested destructuring can get a little hard to follow, so you can also format it differently to make it shorter in terms of line length and to read:
```javascript
handleInputChange = ({
  target: { name, value }
}) => {
  this.setState({
    [name]: value
  })
};
```

Reply:
Wow, these are all very cool! Thanks for showing me these alternate ways!